# Problem Solving and Programming
## Day-13
## Date-11 May 2019

### Day objectives
    *objective 1
    *objective 2
    *objective 3

### Problem 1:

#### Problem Statement
Define a function to check if a given year is a leap year. Returns a boolean value

#### Constraints

### Test cases
    *2000 -> True
    *1900 -> False
    *2012 -> True
    *2020 -> True
    *0200 -> False
    

In [13]:
def leapyear(n):
    if (n%400==0 or n%100!=0):
        return True
    elif(n%4!=0):
        return False
    else:
        return False
leapyear(2012)
        

    

True

### Problem 2:

#### Problem Statement
 Define a function to convert a decimal number to the corresponding binary number
 
#### Constraints

### Test cases
    *decimalToBinary(15) -> 1111
    *decimalToBinary(1) -> 1

In [22]:
def decimalToBinary(n):
    if(n>1):
        decimalToBinary(n//2)
    print(n%2,end='')
decimalToBinary(10)
    

1010

### Problem 3:

#### Problem Statement
Define a function to convert a binary number to the corresponding decimal number

#### Constraints

### Test cases
    *binaryToDecimal(1100) -> 12
    *binaryToDecimal(1010) -> 10
    *binaryToDecimal(111000) -> 56

In [7]:
def binaryToDecimal(n):
    num=n
    dec=0
    base=1
    temp=num
    while(temp):
        rem=temp%10
        temp=temp//10
        dec=dec+base*rem
        base=base*2
    return dec
binaryToDecimal(1000)

8

### Problem 4:

#### Problem Statement
 Define a function to merge the characters of two strings alternatively. The remaining characters

 of the longer string are printed in the same order at the end.
 
#### Constraints

### Test cases
    *mergeString('abcd', 'abcd') -> 'aabbccdd'
    *mergeString('abc', '123456') -> 'a1b2c3456'
    *mergeString('0', '123456') -> '0123456'

In [10]:
def mergeString(s1,s2):
    k=[]
    for count in range (0,len(max(s1,s2))):
        k.append(s1[count]+s2[count])
    if len(s1)<len(s2):
        for count in range (len(s1),len(s2)):
            k.append(s2[count])
    else:
        for count in range (len(s2),len(s1)):
            k.append(s1[count])
    return listmerge(k)
def listmerge(k):
    m=""
    for count in range(0,len(k)):
        m=m+k[count]
    return (m)
print(mergeString('abcd', 'abcd'))
print(mergeString('abc', '123456'))
print(mergeString('0',' 123456'))

aabbccdd
a1b2c3456
0 123456
